multiapi-engine/src/main/java/com/sngular/api/generator/plugin/commonApi/MapperUtilCommon.java

```diff
@@ -0,0 +1,199 @@
+        } else if (DOUBLE.equalsIgnoreCase(format)) {
+          type = DOUBLE;
+        } else {
+          type = BIG_DECIMAL;
+        }
+      } else if (INTEGER.equalsIgnoreCase(type)) {
+        if (INT_64.equalsIgnoreCase(format)) {
+          type = LONG;
+        } else {
+          type = INTEGER;
+        }
+      }
+
+    } else  {
+      type = getPojoName(getRefClass(schema), prefix, suffix);
+    }
+    return StringUtils.capitalize(type);
+  }
+
+  public static String getTextValue(JsonNode node, String field){
+    try{
+      if(checkTextValueNotNull(node.get(field)))
+        return node.get(field).textValue();
+    }
+    catch (StackOverflowError e){
+      return null;
+    }
+    return null;
+  }
+  public static boolean checkTextValueNotNull(JsonNode schema){
+    return (schema != null && schema.textValue()!= null && !schema.isNull());
+  }
+
+  public static String getPojoName(final String namePojo, final String prefix, final String suffix) {
+    return StringUtils.defaultIfBlank(prefix, "")
+            + StringUtils.capitalize(namePojo)
+            + StringUtils.defaultIfBlank(suffix, "");
+  }
+
+  public static String getRefClass(final JsonNode schema) {
+    final String[] pathObjectRef = schema.get(REF).textValue().split(SLASH);
+    return pathObjectRef[pathObjectRef.length - 1];
+  }
+
+  public static String getTypeMap(final JsonNode mapSchema, final String prefix, final String suffix) {
+    var typeMap = "";
+    if (mapSchema.get(ADDITIONAL_PROPERTIES)==null) return null;
+    final var mapNode = mapSchema.get(ADDITIONAL_PROPERTIES);
+    final var mapValueType = mapNode.findPath(TYPE);
+    typeMap = getCollectionType(mapNode, mapValueType, prefix, suffix);
+    return typeMap;
+  }
+
+  public static String getCollectionType(final JsonNode mapNode, final JsonNode mapValueType, final String prefix, final String suffix) {
+    String typeMap = null;
+    if ((checkTextValueNotNull(mapValueType))&&(!mapValueType.textValue().contains("#"))) {
+      if (STRING.equalsIgnoreCase(mapValueType.textValue())) {
+        typeMap = STRING;
+      } else if (INTEGER.equalsIgnoreCase(mapValueType.textValue())) {
+        typeMap = INTEGER;
+      } else {
+        typeMap = mapValueType.textValue();
+      }
+    } else if (mapNode != null){
+      final var valueSchema = mapNode.findPath(REF);
+      if (Objects.nonNull(valueSchema)) {
+        if (checkTextValueNotNull(mapNode.get(REF)))
+          typeMap = getPojoName(getRefClass(mapNode), prefix, suffix);
+      }
+    }
+    return typeMap;
+  }
+
+  public static String getTypeArray(final JsonNode array, final String prefix, final String suffix) {
+    var typeArray = "";
+    final var arrayNode = array.get(ITEMS);
+    final JsonNode mapValueType;
+    if (arrayNode.has(TYPE)) {
+      mapValueType = arrayNode.get(TYPE);
+    } else {
+      mapValueType = arrayNode.get(REF);
+    }
+    typeArray = getCollectionType(arrayNode, mapValueType, prefix, suffix);
```

Comment:
I'm looking this code and L.88 and I think that the condition can be moved inside `getCollectionType` method:
* in both, the second parameter is obtained from the first parameter (in L.88 send always the `TYPE` but in L.116, it check if the `TYPE` is present?)
* the issue is: does this method called from other parts of the code?
* even inside the own method it checks if the second parameter contains `#` (as it would be a reference, instead of a type)